File: neo4jHelloWorld.py
from neo4j.v1 import GraphDatabase, basic_auth
import sys

hostname=sys.argv[1]
username=sys.argv[2]
password=sys.argv[3]

#connect to local Neo4j
driver = GraphDatabase.driver("bolt://" + hostname + ":7687",
    auth=basic_auth(username,password))
session = driver.session()

#create Python language entity
session.run("CREATE (:Language {name:{name},version:{ver}});",
    {"name":"Python","ver":"2.7.13"})

#create relationship from Python lanugage to welcome message
createRelationship = """MATCH (m:Message),(l:Language)
    WHERE m.title = 'Welcome' AND l.name='Python'
    CREATE (m)-[:ACCESSED_FROM]->(l);"""
session.run(createRelationship)

#query the message and the relationship to Python
queryRelationship = """MATCH (m:Message)-[:ACCESSED_FROM]->
    (l:Language {name:'Python'})
    RETURN m,l;"""
resultSet = session.run(queryRelationship)

for result in resultSet:
      print("%s from %s" % (result["m"]["text"], result["l"]["name"]))

session.close()

File: neo4j/Neo4jHelloWorld.java
package neo4jHelloWorld;

import org.neo4j.driver.v1.Session;
import org.neo4j.driver.v1.StatementResult;
import org.neo4j.driver.v1.Record;

import static org.neo4j.driver.v1.Values.parameters;

public class Neo4jHelloWorld {

	public static void main(String[] args) {
		//connect to local Neo4j
		Neo4jConnection conn = new Neo4jConnection("192.168.0.100","neodba","flynnLives");
		Session session = conn.getSession();
		
		//create Java language entity
		session.run( "CREATE (:Language {name:{name},version:{ver}})",
		        parameters("name", "Java", "ver", "1.8.0_74"));

		//create relationship from Java language to welcome message
		String createRelationship = "MATCH (m:Message),(l:Language) "
		    + "WHERE m.title = {title} AND l.name={language} "
		    + "CREATE (m)-[:ACCESSED_FROM]->(l);";
		session.run(createRelationship,
				parameters("title", "Welcome", "language", "Java"));

		//query the message and the relationship to Java
		String queryRelationship = "MATCH (m:Message)-[:ACCESSED_FROM]->"
			+ "(l:Language {name:{language}}) "
		   + "RETURN m.title,l.name;";
		StatementResult resultSet = session.run(queryRelationship, parameters("language", "Java"));

		//process result set
		while (resultSet.hasNext())
		{
		    Record result = resultSet.next();
		    System.out.println( result.get("m.title") + " from " + result.get("l.name"));
		}

		session.close();
	}
}
